### DISCUSSION TITLE
Another new build

### DISCUSSION THREAD

**Discussion by coasty489** [kachurovskiy/megaels#23] [General]
Hi. I’ve been impressed with your Utube videos and openness with information and so have just ordered PCB and components as you suggest. I have printed the cases and am in the process of attaching an encoder and Nema23 closed loop stepper to my minilathe. 
I’d just like to thank you Maxim for making all this information available.
I’m in the UK and will hopefully make updates with my progress.
Mike.

**Comment by kachurovskiy** [comment]
Hi Mike, thank you and good luck with your build! We'd love to see how it turns out.

**Reply by coasty489** [reply]
Thanks Maxim.
I’ve now finished the PCB and enclosure build (it all went very smoothly), but wondered if you or someone could explain where I should earth the shielding on the signal wires from the encoder and joystick/pulser.
This is all new to me, but am learning all the time.
Mike.

**Reply by kachurovskiy** [reply]
@coasty489 you can connect wire shielding to the GND terminal on the PCB. Both encoder and pendant terminals have one GND each.

**Reply by coasty489** [reply]
Ok, I’ll attach the shielding braid to the earth wire in the terminal block connector.
Thanks.

**Reply by coasty489** [reply]
My lathe hardware is now nearly finished, so I have changed parameters in IDE to suit. On verifying I get on line 638, DDRG not declared in this scope.
I have just started learning Arduino code but I don’t understand this.
Anyone able to help please.

**Reply by kachurovskiy** [reply]
@coasty489 try selecting the right board at the top in Arduino IDE - something like "Arduino Mega 2560" - this error should go away as per https://forum.arduino.cc/t/ddre-and-ddrj-not-declared-within-the-scope/153324/5

Reason for the error is that the board you currently have selected doesn't have the hardware registers `DDRG`.

**Reply by coasty489** [reply]
Thankyou Maxim. That did the trick.

**Reply by coasty489** [reply]
Maxim. I’ve been using my Mega ELS for a couple of years but have recently had to change the Arduino board. When verifying the code I get an error. Spindle was not declared in this scope. Possibly line 704. Do you have any idea what might cause this.
Thanks Mike.

**Reply by kachurovskiy** [reply]
@coasty489 please post the entire error message here and include your .ino file if possible, thanks!

**Reply by coasty489** [reply]
Maxim. this is a copy of the error message. The .ino is directly downloaded and unchanged from github although i had to copy and rename it to make it editable.
Thankyou for your time with this and if you had one of your ready built Nanoels systems available, i would be interested in buying.
Regards Mike



C:\Users\steam\OneDrive\Documents\Arduino\editable_els\editable_els.ino: In function 'void loop()':
C:\Users\steam\OneDrive\Documents\Arduino\editable_els\editable_els.ino:704:3: error: 'Spindle' was not declared in this scope
   Spindle();
   ^~~~~~~
C:\Users\steam\OneDrive\Documents\Arduino\editable_els\editable_els.ino:705:3: error: 'Read_ADC_Feed' was not declared in this scope
   Read_ADC_Feed();
   ^~~~~~~~~~~~~
C:\Users\steam\OneDrive\Documents\Arduino\editable_els\editable_els.ino:705:3: note: suggested alternative: 'ADC_Feed'
   Read_ADC_Feed();
   ^~~~~~~~~~~~~
   ADC_Feed
C:\Users\steam\OneDrive\Documents\Arduino\editable_els\editable_els.ino:706:29: error: 'Menu' was not declared in this scope
   if (KEYB_TIMER_FLAG != 0) Menu();
                             ^~~~
C:\Users\steam\OneDrive\Documents\Arduino\editable_els\editable_els.ino:706:29: note: suggested alternative: 'enum'
   if (KEYB_TIMER_FLAG != 0) Menu();
                             ^~~~
                             enum
C:\Users\steam\OneDrive\Documents\Arduino\editable_els\editable_els.ino:708:35: error: invalid cast to abstract class type 'Print'
   if (Mode == Mode_Divider) Print(); // пока для теста !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
                                   ^
In file included from C:\Users\steam\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.8.8\cores\arduino/Stream.h:26:0,
                 from C:\Users\steam\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.8.8\cores\arduino/HardwareSerial.h:29,
                 from C:\Users\steam\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.8.8\cores\arduino/Arduino.h:233,
                 from C:\Users\steam\AppData\Local\arduino\sketches\235890E3FCE081812D56E09329936017\sketch\editable_els.ino.cpp:1:
C:\Users\steam\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.8.8\cores\arduino/Print.h:37:7: note:   because the following virtual functions are pure within 'Print':
 class Print
       ^~~~~
C:\Users\steam\AppData\Local\Arduino15\packages\arduino\hardware\avr\1.8.8\cores\arduino/Print.h:51:20: note: 	virtual size_t Print::write(uint8_t)
     virtual size_t write(uint8_t) = 0;
                    ^~~~~
C:\Users\steam\OneDrive\Documents\Arduino\editable_els\editable_els.ino: In function 'void setup()':
C:\Users\steam\OneDrive\Documents\Arduino\editable_els\editable_els.ino:1346:6: error: redefinition of 'void setup()'
 void setup() {
      ^~~~~
C:\Users\steam\OneDrive\Documents\Arduino\editable_els\editable_els.ino:631:6: note: 'void setup()' previously defined here
 void setup()
      ^~~~~
C:\Users\steam\OneDrive\Documents\Arduino\editable_els\editable_els.ino: In function 'void loop()':
C:\Users\steam\OneDrive\Documents\Arduino\editable_els\editable_els.ino:1351:6: error: redefinition of 'void loop()'
 void loop() {
      ^~~~
C:\Users\steam\OneDrive\Documents\Arduino\editable_els\editable_els.ino:702:6: note: 'void loop()' previously defined here
 void loop()
      ^~~~
exit status 1

Compilation error: 'Spindle' was not declared in this scope

**Reply by kachurovskiy** [reply]
The problem is likely caused by partial copying or renaming, Arduino needs all the files in the `Digital_Feed_7e2` folder as-is for the sketch to work.

Try extracting `Digital_Feed_7e2` folder from this repo ZIP file into a new location. Don't forget to install LiquidCrystal library.

**Comment by coasty489** [comment]
Thanks Maxim for your support and time. I think I’m moving forward now. Kind regards Mike. Sent from my iPhoneOn 26 Sep 2026, at 12:22, Maxim Kachurovskiy ***@***.***> wrote:﻿
The problem is likely caused by partial copying or renaming, Arduino needs all the files in the Digital_Feed_7e2 folder as-is for the sketch to work.
Try extracting Digital_Feed_7e2 folder from this repo ZIP file into a new location. Don't forget to install LiquidCrystal library.
